### PR TITLE
Improve PyGMT documentation

### PR DESCRIPTION
**Pull request by core-man**
**Description of proposed changes**

Related to #983.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: doc/install.rst [hunk 1]
```diff
@@ -142,6 +147,10 @@ This installs the latest stable release from
 
     pip install pygmt
 
+This upgrades installed PyGMT to be the latest stable release::
+
+    pip install --upgrade pygmt
+
 Alternatively, you can install the latest development version from
```
Comment (core-man) [inline doc/install.rst:154]: I am still testing how to upgrade the latest dev version.

Comment (willschlitzer) [reply doc/install.rst:154]: The way I have been doing it is cloning the repository and setting up a second conda environment that doesn't have pygmt installed. I changed into the pygmt folder in the repository, and ran `pip install -e`. The pygmt "version" changes depending on which branch I have checked out. If it's the master branch, it is the latest dev version. It also allows me to change into a different branch to test out a new wrapped function, error message, etc.

Comment (core-man) [reply doc/install.rst:154]: @willschlitzer yes, your method is similar to the way I do. But the users have to clone the repo themselves.

---

I used `TestPyPI` to install `0.3.1.dev51` several hours ago:
```
pip install --pre --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pygmt
```

Now I ran the command again. the dev pygmt is not updated and is still `0.3.1.dev51`.

---

I also used install `0.3.1.dev51` from PyGMT’s GitHub repository several hours ago:
```
pip install git+https://github.com/GenericMappingTools/pygmt.git#egg=pygmt
```
Now I ran the command again, and the pygmt is updated to be `0.3.1.dev52`.
```
Collecting pygmt
  Cloning https://github.com/GenericMappingTools/pygmt.git to /tmp/pip-install-r9g_6l77/pygmt_bb7f2501a6bc48b49048fd305155daf0
  Running command git clone -q https://github.com/GenericMappingTools/pygmt.git /tmp/pip-install-r9g_6l77/pygmt_bb7f2501a6bc48b49048fd305155daf0
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
...
...
Installing collected packages: pygmt
  Attempting uninstall: pygmt
    Found existing installation: pygmt 0.3.1.dev51+g10482f3
    Uninstalling pygmt-0.3.1.dev51+g10482f3:
      Successfully uninstalled pygmt-0.3.1.dev51+g10482f3
Successfully installed pygmt-0.3.1.dev52+g55c6176
```

Therefore, I think we may use `pip install git+https://github.com/GenericMappingTools/pygmt.git#egg=pygmt` to install and upgrade dev PyGMT and don't have to clone the repo ourselves.

File: examples/tutorials/first-figure.py [hunk 2]
```diff
@@ -34,11 +35,12 @@
...
-# Add elements to the figure using its methods. For example, let's start a map with an
-# automatic frame and ticks around a given longitude and latitude bound, set the
-# projection to Mercator (``M``), and the map width to 8 inches:
+# Add elements to the figure using its methods. For example, let's use
+# :meth:`pygmt.Figure.basemap` to start a map for a region indicated by a given
+# longitude and latitude bound, set the projection to Mercator (``M``), the
+# map width to 8 centimeters, and frame type to be generated automatically:
 
-fig.basemap(region=[-90, -70, 0, 20], projection="M8i", frame=True)
+fig.basemap(region=[-90, -70, 0, 20], projection="M8c", frame=True)
```
Comment (core-man) [inline examples/tutorials/first-figure.py:43]: Is it better to use SI?

File: examples/tutorials/first-figure.py [hunk 3]
```diff
@@ -80,7 +82,7 @@
 # 1. The name of method is ``coast`` instead of ``pscoast``. As a general rule, all
 #    ``ps*`` modules had their ``ps`` prefix removed. The exceptions are:
 #    ``psxy`` which is now ``plot``, ``psxyz`` which is now ``plot3d``, and ``psscale``
-#    which is now ``colorbar``.
+#    which is now ``colorbar``. Those reflect the new GMT modern execution mode.
```
Comment (core-man) [inline examples/tutorials/first-figure.py:85]: It seems that only point 1 reflects the differences between classic GMT and GMT modern mode, while points 2-5 are PyGMT exclusive. Shall we divide 1-5 into two parts and also update the introduction? One discusses the differences between classic GMT and GMT modern mode, and the other is about PyGMT.

File: examples/tutorials/first-figure.py [hunk 4]
```diff
@@ -7,7 +7,8 @@
 
 .. note::
 
-    This tutorial assumes the use of a Python notebook, such as IPython or Jupyter Notebook.
+    This tutorial assumes the use of a Python notebook, such as `IPython <https://www.pygmt.org/latest/install.html#using-pip>`__
```
Comment (core-man) [inline examples/tutorials/first-figure.py:10]: Is IPython a Python notebook? or is it notebook server/kernel?

**Comment by core-man** [comment]
Is it good to submit revisions for different `.rst` files in this single PR? Now I made some revisions for `install.rst` and `first-figure.rst`. But based on [changelog](https://www.pygmt.org/dev/changes.html), I found you update each documentation in a single PR.

File: examples/tutorials/first-figure.py (hunk 4 above)
Comment (willschlitzer) [reply examples/tutorials/first-figure.py:10]: I was under the impression that it is a notebook. But I've only ever used Jupyter.

**Comment by willschlitzer** [comment]
> @GenericMappingTools/python Is it good to submit revisions for different `.rst` files in this single PR? Now I made some revisions for `install.rst` and `first-figure.rst`. But based on [changelog](https://www.pygmt.org/dev/changes.html), I found you update each documentation in a single PR.

It's situation dependent and a matter of opinion. In my opinion, if you're making big changes to the files, probably best to split it into multiple PRs so all of the file changes aren't dependent on a single PR getting approved. But if it's relatively minor changes (formatting, spelling, adding a single line/description to multiple files) I think they should be grouped together in a single PR.

File: doc/install.rst [hunk 5]
```diff
@@ -51,9 +51,10 @@ work, but there is no guarantee that PyGMT will behave as expected.
 
 We recommend using the `Anaconda <https://www.anaconda.com/distribution>`__
 Python distribution to ensure you have all dependencies installed and the
-``conda`` package manager available.
-Installing Anaconda does not require administrative rights to your computer and
-doesn't interfere with any other Python installations in your system.
+`conda <https://docs.conda.io/projects/conda/en/latest/user-guide/index.html>`__
```
Comment (seisman) [inline doc/install.rst:54]:
```suggestion
`conda <https://docs.conda.io/projects/conda/en/latest/>`__
```

File: doc/install.rst [hunk 6]
```diff
@@ -142,6 +147,10 @@ This installs the latest stable release from
 
     pip install pygmt
 
+This upgrades installed PyGMT to be the latest stable release::
```
Comment (meghanrjones) [inline doc/install.rst:150]:
```suggestion
This upgrades the installed PyGMT to be the latest stable release::
```

File: doc/install.rst (hunk 1 above)
Comment (seisman) [reply doc/install.rst:154]: If you use the following command to install the dev version,
```
pip install --pre --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pygmt
```

you can simply add `--upgrade` to update pygmt.

Comment (seisman) [reply doc/install.rst:154]: > I think we may use `pip install git+https://github.com/GenericMappingTools/pygmt.git#egg=pygmt` to install and upgrade dev PyGMT and don't have to clone the repo ourselves.

It's much slower than the `pip install --pre` method and we show both ways in the installing guide.

File: doc/install.rst [hunk 7]
```diff
@@ -152,8 +161,8 @@ or from PyGMT's `GitHub repository <https://github.com/GenericMappingTools/pygmt
 
     pip install git+https://github.com/GenericMappingTools/pygmt.git#egg=pygmt
 
-Any of the above methods (conda/pip) should allow you to use the ``pygmt``
-library from Python.
+Any of the above methods (conda/pip) should allow you to use the PyGMT library
+from Python.
```
Comment (seisman) [inline doc/install.rst:165]: I'm wondering should we call PyGMT a "library" or a "package".

File: examples/tutorials/first-figure.py (hunk 4 above)
Comment (seisman) [reply examples/tutorials/first-figure.py:10]: If I understand it correctly, IPython is a console and also a kernel, and Jupyter Notebook use the IPython kernel to work.

Comment (seisman) [inline examples/tutorials/first-figure.py:10]:
```suggestion
    This tutorial assumes the use of a Python notebook, such as `IPython <https://ipython.org/>`__
```

File: examples/tutorials/first-figure.py (hunk 2 above)
Comment (seisman) [reply examples/tutorials/first-figure.py:43]: Yes, we should use SI units.

File: doc/install.rst (hunk 7 above)
Comment (core-man) [reply doc/install.rst:165]: I think it is a package.

File: examples/tutorials/first-figure.py (hunk 4 above)
Comment (core-man) [reply examples/tutorials/first-figure.py]: Do anyone just install IPython for PyGMT without Jupyter Notebook or JupyterLAB?

Comment (seisman) [reply examples/tutorials/first-figure.py]: We don't make any assumptions. I think it's common that people have ipython, but not jupyter notebook.

File: examples/tutorials/first-figure.py (hunk 2 above)
Comment (weiji14) [reply examples/tutorials/first-figure.py:43]: There's another M8i at Line 61 to change.

File: doc/install.rst [hunk 8]
```diff
@@ -152,8 +161,8 @@ or from PyGMT's `GitHub repository <https://github.com/GenericMappingTools/pygmt
 
     pip install git+https://github.com/GenericMappingTools/pygmt.git#egg=pygmt
 
-Any of the above methods (conda/pip) should allow you to use the ``pygmt``
-library from Python.
+Any of the above methods (conda/pip) should allow you to use the PyGMT library
```
Comment (seisman) [inline doc/install.rst:164]:
```suggestion
Any of the above methods (conda/pip) should allow you to use the PyGMT package
```

File: examples/tutorials/first-figure.py (hunk 2 above)
Comment (core-man) [reply examples/tutorials/first-figure.py:43]: > There's another M8i at Line 61 to change.

Revised in 5f67156.

File: doc/install.rst (hunk 1 above)
Comment (core-man) [reply doc/install.rst]: How to upgrade the dev version was added in 2c730e5.

File: examples/tutorials/first-figure.py [hunk 9]
```diff
@@ -34,11 +35,12 @@
 fig = pygmt.Figure()
 
 ########################################################################################
-# Add elements to the figure using its methods. For example, let's start a map with an
-# automatic frame and ticks around a given longitude and latitude bound, set the
-# projection to Mercator (``M``), and the map width to 8 inches:
+# Add elements to the figure using its methods. For example, let's use
+# :meth:`pygmt.Figure.basemap` to start a map for a region indicated by a given
+# longitude and latitude bound, set the projection to Mercator (``M``), the
```
Comment (seisman) [inline examples/tutorials/first-figure.py:40]:
```suggestion
# longitude and latitude bound, set the projection to Mercator (**M**), the
```

File: examples/tutorials/first-figure.py [hunk 10]
```diff
@@ -74,20 +76,25 @@
...
+# command-line GMT. Many of these changes reflect the new GMT modern execution
+# mode that are part of GMT 6.
 #
 # 1. The name of method is ``coast`` instead of ``pscoast``. As a general rule, all
 #    ``ps*`` modules had their ``ps`` prefix removed. The exceptions are:
 #    ``psxy`` which is now ``plot``, ``psxyz`` which is now ``plot3d``, and ``psscale``
 #    which is now ``colorbar``.
-# 2. The parameters don't use the GMT 1-letter syntax (**R**, **J**, **B**, etc). We use longer
+#
+# 2. More details can be found at https://docs.generic-mapping-tools.org/latest/cookbook/introduction.html?highlight=modern#modern-and-classic-mode
```
Comment (seisman) [inline examples/tutorials/first-figure.py:87]:
```suggestion
# 2. More details can be found at :gmt-docs:`/cookbook/introduction.html#modern-and-classic-mode`
```

File: examples/gallery/plot/inset.py [hunk 11]
```diff
@@ -5,7 +5,7 @@
 The :meth:`pygmt.Figure.inset` method adds an inset figure inside a larger
 figure. The function is called using a ``with`` statement, and its position,
 box, offset, and margin parameters are set. Within the ``with`` statement,
-PyGMT plotting functions can be called that add to the inset figure.
+PyGMT plotting functions can be called to plot the inset figure.
```
Comment (core-man) [inline examples/gallery/plot/inset.py:8]: @willschlitzer Is my revision right?

Comment (liamtoney) [reply examples/gallery/plot/inset.py:8]: I suggest the following sentence instead:

> `Plotting methods called within the ``with`` statement plot into the inset figure.`

File: examples/tutorials/first-figure.py [hunk 12]
```diff
@@ -34,11 +35,12 @@
 fig = pygmt.Figure()
 
 ########################################################################################
-# Add elements to the figure using its methods. For example, let's start a map with an
-# automatic frame and ticks around a given longitude and latitude bound, set the
-# projection to Mercator (``M``), and the map width to 8 inches:
+# Add elements to the figure using its methods. For example, let's use
+# :meth:`pygmt.Figure.basemap` to start a map for a region indicated by a given
+# longitude and latitude bound, set the projection to Mercator (**M**), the
+# map width to 8 centimeters, and frame type to be generated automatically:
```
Comment (michaelgrund) [inline examples/tutorials/first-figure.py:41]:
```suggestion
# map width to 8 cm, and frame type to be generated automatically:
```
would use the short form cm here

File: examples/tutorials/first-figure.py [hunk 13]
```diff
@@ -74,20 +76,25 @@
 # ------------------------------
 #
 # You’ll probably have noticed several things that are different from classic
-# command-line GMT. Many of these changes reflect the new GMT modern execution mode that
-# are part of GMT 6. A few are PyGMT exclusive (like the ``savefig`` method).
+# command-line GMT. Many of these changes reflect the new GMT modern execution
+# mode that are part of GMT 6.
```
Comment (michaelgrund) [inline examples/tutorials/first-figure.py:80]:
```suggestion
# mode that is part of GMT 6.
```

File: examples/gallery/plot/inset.py (hunk 11 above)
Comment (core-man) [reply examples/gallery/plot/inset.py]: Done in b260f45.

File: doc/install.rst [hunk 14]
```diff
@@ -142,18 +147,28 @@ This installs the latest stable release from
...
+    pip install --upgrade --pre --index-url https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple pygmt
+
+Alternatively, you can install the latest development version from PyGMT's `GitHub repository <https://github.com/GenericMappingTools/pygmt>`__ 
 (slower as it downloads the whole archive)::
 
     pip install git+https://github.com/GenericMappingTools/pygmt.git#egg=pygmt
 
-Any of the above methods (conda/pip) should allow you to use the ``pygmt``
-library from Python.
+Run the above command again to upgrade the latest development.
```
Comment (seisman) [inline doc/install.rst:168]:
```suggestion
Run the above command again to upgrade to the latest development version.
```

File: examples/gallery/line/linestyles.py [hunk 15]
```diff
@@ -38,7 +38,7 @@
     "1p,lightblue,-.",  # dash-dotted line
     "2p,blue,..-",  # dot-dot-dashed line
     "2p,tomato,--.",  # dash-dash-dotted line
-    "2p,tomato,4_2:2p",  # A pattern of 4-point-long line segment and 2-point-gap between segment
+    "2p,tomato,4_2:2p",  # A pattern of 4-point-long line segment and 2-point-gap between segment with pattern offset by 2 points from the origin
```
Comment (seisman) [inline examples/gallery/line/linestyles.py:41]: Avoid long lines.
```suggestion
    # A pattern of 4-point-long line segment and 2-point-gap between segment,
    # with pattern offset by 2 points from the origin
    "2p,tomato,4_2:2p",  
```

**Comment by weiji14** [comment]
@core-man, I appreciate all the improvements you're making here, but this Pull Request is getting a bit large in scope. The general guidelines at https://github.com/GenericMappingTools/pygmt/blob/v0.3.0/CONTRIBUTING.md#general-guidelines recommend that:

- Each pull request should consist of a **small** and logical collection of changes.
- Larger changes should be broken down into smaller components and integrated separately.

In particular, the changes made to the gallery examples might cause some tricky merge conflicts with the gallery reorganization at #995. I suggest you coordinate a bit with @willschlitzer on this matter before things get more complicated.

**Comment by core-man** [comment]
> @core-man, I appreciate all the improvements you're making here, but this Pull Request is getting a bit large in scope. The general guidelines at https://github.com/GenericMappingTools/pygmt/blob/v0.3.0/CONTRIBUTING.md#general-guidelines recommend that:
> 
> * Each pull request should consist of a **small** and logical collection of changes.
> * Larger changes should be broken down into smaller components and integrated separately.

Thanks for the suggestion. I agree this PR is a bit large in scope, although I 
plan to separate some other revisions (e.g.,  https://github.com/GenericMappingTools/pygmt/issues/983#issuecomment-791177253, https://github.com/GenericMappingTools/pygmt/issues/983#issuecomment-791920354) in other PRs.

> In particular, the changes made to the gallery examples might cause some tricky merge conflicts with the gallery reorganization at #995. I suggest you coordinate a bit with @willschlitzer on this matter (i.e. decide who merges their PR first) before things get more complicated.

yes, I also notice #995 which I think should be merged first.

**Comment by core-man** [comment]
Based on the suggestion by @weiji14, I will re-submit all the revisions in this PR by a few small PRs. Each small PR only contain revisions of 1~2 files. #1025 is the first one. Sorry for the inconvenience.

**Comment by seisman** [comment]
@core-man Do you want to apply some changes in this PR before v0.3.1? If you do, please open separate PRs for them.

**Comment by core-man** [comment]
> @core-man Do you want to apply some changes in this PR before v0.3.1? If you do, please open separate PRs for them.

Thanks, I'd like to apply some after v0.3.1.

**Comment by core-man** [comment]
Close this PR since the revisions have been submitted by the individual PRs.

**Pull request closed without merging**